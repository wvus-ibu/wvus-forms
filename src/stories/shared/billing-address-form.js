import React, { Component } from "react";
import {
  WVUSForm,
  CheckboxControl,
  SelectControl,
  InputControl,
  validateRequired,
  validateEmail,
  validatePhone,
  validateZip,
  validateMin,
  validateMax,
  validateEmpty,
  USStateFieldOptions
} from "../../index";

class BillingInfoForm extends Component {
  constructor(props) {
    super(props);
    this.handleFormSubmit = this.handleFormSubmit.bind(this);
  }

  handleFormSubmit(e) {
    // Handle your form submission however you would like
    e.preventDefault();
    // fail safe to bail if they get around disabled button
    if (!this.props.formMethods.isFormValid()) return false;
  }

  render() {
    const { formMethods } = this.props;
    const submitDisabled = !formMethods.isFormValid();
    return (
      <form>
        <h2>Billing Info Form</h2>
        <InputControl
          labelClasses="first-name-label"
          inputClasses="first-name-input"
          fieldName="fname"
          fieldTitle="First Name"
          fieldClasses="wvus-field-50"
          formMethods={formMethods}
          validators={[validateRequired]}
        />

        <InputControl
          fieldName="lname"
          fieldTitle="Last Name"
          fieldClasses="wvus-field-50 wvus-field--end"
          formMethods={formMethods}
          validators={[validateRequired]}
        />

        <InputControl
          fieldName="address1"
          fieldTitle="Mailing Address Line 1"
          formMethods={formMethods}
          validators={[validateRequired]}
        />

        <InputControl
          fieldName="address2"
          fieldTitle="Mailing Address Line 2"
          optional={true}
          isValid={true}
          secondInteraction={false}
          formMethods={formMethods}
          validators={[]}
        />

        <InputControl
          fieldName="city"
          fieldTitle="City"
          fieldClasses="wvus-field-33"
          formMethods={formMethods}
          validators={[validateRequired]}
        />

        <SelectControl
          fieldName="state"
          fieldOptions={USStateFieldOptions}
          fieldTitle="State"
          fieldClasses="wvus-field-33"
          formMethods={formMethods}
          validators={[validateRequired, validateMin(2)]}
        />

        <InputControl
          fieldName="zip"
          fieldTitle="Zip"
          fieldClasses="wvus-field-33 wvus-field--end"
          formMethods={formMethods}
          validators={[validateRequired, validateZip]}
        />
        <InputControl
          fieldName="email"
          fieldTitle="Email"
          fieldType="email"
          fieldClasses="wvus-field-newletter"
          formMethods={formMethods}
          validators={[validateRequired, validateEmail]}
        />

        <InputControl
          fieldName="phone"
          fieldTitle="Phone"
          fieldType="tel"
          fieldClasses="wvus-field-newletter"
          formMethods={formMethods}
          validators={[validateRequired, validatePhone]}
        />
        <CheckboxControl
          fieldName="optIn"
          fieldValue={true}
          fieldTitle="Yes, please add me to your email newsletter"
          labelClasses="optIn-label"
          inputClasses="optIn-input"
          fieldClasses="wvus-field-newletter-optin"
          formMethods={formMethods}
        />

        <button
          type="submit"
          id="wv-login-form-submit"
          className="btn btn-medium btn-secondary"
          disabled={submitDisabled}
          onClick={this.handleFormSubmit}
        >
          Submit
        </button>
      </form>
    );
  }
}
export { BillingInfoForm };
