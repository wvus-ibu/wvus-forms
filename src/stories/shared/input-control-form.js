import React from "react";
import { action } from "@storybook/addon-actions";

import {
  WVUSForm,
  InputControl,
  SelectControl,
  CheckboxControl
} from "../../index";
import {
  validateRequired,
  validateEmail,
  validatePhone,
  validateZip,
  validateMin,
  validateMax
} from "../../index";

const formMethodsUnTouched = {
  addFieldToState: () => {},
  getFieldState: () => {
    return {
      value: "",
      secondInteraction: false,
      isValid: true,
      optional: false,
      errorMessage: "Error message displays here"
    };
  },
  handleBlur: action("handleblur"),
  handleValueChange: action("handleValueChange"),
  showUIError: () => false,
  showUISuccess: () => false
};
const propsUntouched = {
  formMethods: formMethodsUnTouched
};
const formMethodsWithSuccess = {
  addFieldToState: () => {},
  getFieldState: () => {
    return {
      value: "Scott Rees",
      secondInteraction: true,
      isValid: true,
      optional: false,
      errorMessage: "Error message displays here"
    };
  },
  handleBlur: action("handleblur"),
  handleValueChange: action("handleValueChange"),
  showUIError: () => false,
  showUISuccess: () => true
};

const propsWithSuccess = {
  formMethods: formMethodsWithSuccess
};

const formMethodsWithError = {
  addFieldToState: () => {},
  getFieldState: () => {
    return {
      value: "",
      secondInteraction: true,
      isValid: false,
      optional: false,
      errorMessage: "Error message displays here"
    };
  },
  handleBlur: action("handleblur"),
  handleValueChange: action("handleValueChange"),
  showUIError: () => true,
  showUISuccess: () => false
};
const propsWithError = {
  formMethods: formMethodsWithError
};

const SimpleInputForm = props => {
  return (
    <form>
      <InputControl
        fieldPlaceholder="Tim Stehlin"
        fieldName="fname"
        fieldTitle="First Name"
        fieldClasses="custom-class-field-input"
        formMethods={props.formMethods}
        validation={[validateRequired]}
      />
    </form>
  );
};

const SimpleInputStates = props => {
  return (
    <form>
      <InputControl
        fieldPlaceholder="Tim Stehlin"
        fieldName="fname"
        fieldTitle="First Name"
        fieldClasses="custom-class-field-input"
        formMethods={props.formMethods}
      />
    </form>
  );
};

export {
  SimpleInputForm,
  SimpleInputStates,
  propsUntouched,
  propsWithError,
  propsWithSuccess
};
