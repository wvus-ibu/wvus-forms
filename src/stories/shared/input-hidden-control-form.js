import React from "react";

import { WVUSForm, InputHiddenControl } from "../../index";

const SimpleInputHiddenForm = props => {
  return (
    <form>
      <InputHiddenControl
        fieldName="campaign"
        fieldValue={1234567}
        fieldClasses="test-hidden-field"
        formMethods={props.formMethods}
      />
      <p>
        Hidden Field Value:{" "}
        {props.formMethods.getFieldState("campaign").value
          ? props.formMethods.getFieldState("campaign").value
          : ""}
      </p>
    </form>
  );
};

export { SimpleInputHiddenForm };
