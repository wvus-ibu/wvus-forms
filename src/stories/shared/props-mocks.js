import { action } from "@storybook/addon-actions";

const formMethodsUnTouched = {
  addFieldToState: () => {},
  getFieldState: () => {
    return {
      value: "",
      secondInteraction: false,
      isValid: true,
      optional: false,
      errorMessage: "Error message displays here"
    };
  },
  handleBlur: action("handleblur"),
  handleValueChange: action("handleValueChange"),
  showUIError: () => false,
  showUISuccess: () => false
};

const formMethodsWithSuccess = {
  addFieldToState: () => {},
  getFieldState: () => {
    return {
      value: "Scott Rees",
      secondInteraction: true,
      isValid: true,
      optional: false,
      errorMessage: "Error message displays here"
    };
  },
  handleBlur: action("handleblur"),
  handleValueChange: action("handleValueChange"),
  showUIError: () => false,
  showUISuccess: () => true
};

const formMethodsWithError = {
  addFieldToState: () => {},
  getFieldState: () => {
    return {
      value: "",
      secondInteraction: true,
      isValid: false,
      optional: false,
      errorMessage: "Error message displays here"
    };
  },
  handleBlur: action("handleblur"),
  handleValueChange: action("handleValueChange"),
  showUIError: () => true,
  showUISuccess: () => false
};

/**
 * Optional fields should have these defaults:
 * isValid: true // allows form to be submitted without a value (since field is optional)
 * optional: true // ensures success state is not shown if field is empty and interacted with (see showUiSuccess function)
 */
const formMethodsOptional = {
  addFieldToState: () => {},
  getFieldState: () => {
    return {
      value: "",
      secondInteraction: true,
      isValid: true,
      optional: true,
      errorMessage: "Error message displays here"
    };
  },
  handleBlur: action("handleblur"),
  handleValueChange: action("handleValueChange"),
  showUIError: () => false,
  showUISuccess: () => false
};

/**
 * Checkbox props
 */

const formMethodsChecked = {
  addFieldToState: () => {},
  getFieldState: () => {
    return {
      value: true,
      secondInteraction: false,
      isValid: true,
      optional: false,
      errorMessage: "Error message displays here"
    };
  },
  handleBlur: action("handleblur"),
  handleValueChange: action("handleValueChange"),
  showUIError: () => false,
  showUISuccess: () => false
};

const formMethodsUnChecked = {
  addFieldToState: () => {},
  getFieldState: () => {
    return {
      value: false,
      secondInteraction: false,
      isValid: true,
      optional: false,
      errorMessage: "Error message displays here"
    };
  },
  handleBlur: action("handleblur"),
  handleValueChange: action("handleValueChange"),
  showUIError: () => false,
  showUISuccess: () => false
};

export {
  formMethodsUnTouched,
  formMethodsWithError,
  formMethodsWithSuccess,
  formMethodsOptional,
  formMethodsChecked,
  formMethodsUnChecked
};
