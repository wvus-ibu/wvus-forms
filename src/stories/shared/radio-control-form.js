import React from "react";

import { WVUSForm, RadioControl } from "../../index";

const RadioForm = props => {
  return (
    <form>
      <RadioControl
        fieldId="payment_method_new_cc"
        fieldName="payment_method"
        fieldTitle={"New Credit Card"}
        formMethods={props.formMethods}
        fieldCheckedDefault={true}
      />
      <RadioControl
        fieldId="payment_method_existing_cc"
        fieldName="payment_method"
        fieldTitle={"Existing Credit Card"}
        formMethods={props.formMethods}
      />
      <RadioControl
        fieldId="payment_method_paypal"
        fieldName="payment_method"
        fieldTitle={"Paypal"}
        formMethods={props.formMethods}
      />
    </form>
  );
};

export { RadioForm };
