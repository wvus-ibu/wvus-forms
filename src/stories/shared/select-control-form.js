import React from "react";

import { WVUSForm, SelectControl, validateRequired } from "../../index";
// import { formMethodsChecked, formMethodsUnChecked } from "./props-mocks";
import USStateFieldOptions from "../../misc/us-states";

const SelectForm = props => {
  return (
    <form>
      <SelectControl
        fieldName="state"
        fieldOptions={USStateFieldOptions}
        fieldTitle="State"
        fieldClasses="wvus-field-33"
        formMethods={props.formMethods}
        validators={[validateRequired]}
      />
    </form>
  );
};

const SelectFormPreselected = props => {
  return (
    <form>
      <SelectControl
        fieldName="state"
        fieldValue="WA"
        fieldOptions={USStateFieldOptions}
        fieldTitle="State"
        isValid={true}
        fieldClasses="wvus-field-33"
        formMethods={props.formMethods}
        validators={[validateRequired]}
      />
    </form>
  );
};

const SelectFormNoConfig = props => {
  return (
    <form>
      <SelectControl
        fieldName="state"
        fieldOptions={props.fieldOptions}
        fieldTitle="State"
        fieldClasses="wvus-field-33"
        formMethods={props.formMethods}
        validators={[validateRequired]}
      />
    </form>
  );
};

export { SelectForm, SelectFormPreselected, SelectFormNoConfig };
