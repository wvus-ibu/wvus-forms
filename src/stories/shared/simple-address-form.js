import React from "react";

import {
  WVUSForm,
  InputControl,
  SelectControl,
  CheckboxControl
} from "../../index";
import {
  validateRequired,
  validateEmail,
  validatePhone,
  validateZip,
  validateMin,
  validateMax
} from "../../index";
import USStateFieldOptions from "../../misc/us-states";

const SimpleAddressForm = props => {
  return (
    <form>
      <h2>Simple Address Form</h2>
      <InputControl
        labelClasses="first-name-label"
        inputClasses="first-name-input"
        fieldName="fname"
        fieldTitle="First Name"
        fieldClasses="wvus-field-50"
        formMethods={props.formMethods}
        validators={[validateRequired]}
      />

      <InputControl
        fieldName="lname"
        fieldTitle="Last Name"
        fieldClasses="wvus-field- wvus-field--end"
        formMethods={props.formMethods}
        validators={[validateRequired]}
      />

      <InputControl
        fieldName="address1"
        fieldTitle="Mailing Address"
        formMethods={props.formMethods}
        validators={[validateRequired]}
      />

      <InputControl
        fieldName="city"
        fieldTitle="City"
        fieldClasses="wvus-field-33"
        formMethods={props.formMethods}
        validators={[validateRequired]}
      />

      <SelectControl
        fieldName="state"
        fieldOptions={USStateFieldOptions}
        fieldTitle="State"
        fieldClasses="wvus-field-33"
        formMethods={props.formMethods}
        validators={[validateRequired, validateMin(2)]}
      />
      <InputControl
        fieldName="zip"
        fieldTitle="Zip"
        fieldClasses="wvus-field-33 wvus-field--end"
        formMethods={props.formMethods}
        validators={[validateRequired, validateZip]}
      />
    </form>
  );
};
export { SimpleAddressForm };
