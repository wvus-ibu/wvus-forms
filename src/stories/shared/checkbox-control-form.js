import React from "react";

import { WVUSForm, CheckboxControl } from "../../index";
import { formMethodsChecked, formMethodsUnChecked } from "./props-mocks";

const propsChecked = {
  formMethods: formMethodsChecked
};

const propsUnchecked = {
  formMethods: formMethodsUnChecked
};

const propsDisabled = {
  formMethods: formMethodsChecked,
  attributes: {
    disabled: true
  }
};

const SimpleCheckboxForm = props => {
  return (
    <form>
      <CheckboxControl
        labelClasses="opt-in-label"
        inputClasses="opt-in-input"
        fieldName="optIn"
        fieldValue={true}
        fieldTitle="Yes, I would like to receive the email newsletter."
        fieldClasses="wvus-field-newletter-optin"
        formMethods={props.formMethods}
      />
    </form>
  );
};

const DisabledCheckboxForm = props => {
  return (
    <form>
      <fieldset disabled>
        <CheckboxControl
          fieldName="optIn"
          fieldValue={true}
          fieldTitle="Yes, I would like to receive the email newsletter."
          fieldClasses="wvus-field-newletter-optin"
          formMethods={props.formMethods}
        />
      </fieldset>
    </form>
  );
};

export {
  SimpleCheckboxForm,
  DisabledCheckboxForm,
  propsChecked,
  propsUnchecked,
  propsDisabled
};
