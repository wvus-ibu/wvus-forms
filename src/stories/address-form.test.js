import React from "react";
import renderer from "react-test-renderer";

import { SimpleAddressForm } from "./shared/simple-address-form";
import { WVUSForm } from "../index";
const Form = WVUSForm(SimpleAddressForm);

test("Simple Address Form renders correctly", () => {
  const tree = renderer.create(<Form />).toJSON();
  expect(tree).toMatchSnapshot();
});
