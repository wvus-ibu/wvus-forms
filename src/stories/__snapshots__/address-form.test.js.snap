// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple Address Form renders correctly 1`] = `
<div
  className=""
>
  <form>
    <h2>
      Simple Address Form
    </h2>
    <div
      className="wvus-field-fname wvus-field-50 form-group has-icon "
      data-field-container="fname"
      data-field-is-valid={false}
    >
      <label
        htmlFor="fname"
      >
        First Name
         
        <sup>
          *
        </sup>
      </label>
      <div
        className="form-control-wrapper"
      >
        <input
          className="form-control"
          id="fname"
          name="fname"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder={undefined}
          type="text"
          value=""
        />
        
        
        
      </div>
    </div>
    <div
      className="wvus-field-lname wvus-field- wvus-field--end form-group has-icon "
      data-field-container="lname"
      data-field-is-valid={false}
    >
      <label
        htmlFor="lname"
      >
        Last Name
         
        <sup>
          *
        </sup>
      </label>
      <div
        className="form-control-wrapper"
      >
        <input
          className="form-control"
          id="lname"
          name="lname"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder={undefined}
          type="text"
          value=""
        />
        
        
        
      </div>
    </div>
    <div
      className="wvus-field-address1  form-group has-icon "
      data-field-container="address1"
      data-field-is-valid={false}
    >
      <label
        htmlFor="address1"
      >
        Mailing Address
         
        <sup>
          *
        </sup>
      </label>
      <div
        className="form-control-wrapper"
      >
        <input
          className="form-control"
          id="address1"
          name="address1"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder={undefined}
          type="text"
          value=""
        />
        
        
        
      </div>
    </div>
    <div
      className="wvus-field-city wvus-field-33 form-group has-icon "
      data-field-container="city"
      data-field-is-valid={false}
    >
      <label
        htmlFor="city"
      >
        City
         
        <sup>
          *
        </sup>
      </label>
      <div
        className="form-control-wrapper"
      >
        <input
          className="form-control"
          id="city"
          name="city"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder={undefined}
          type="text"
          value=""
        />
        
        
        
      </div>
    </div>
    <div
      className="wvus-field-state wvus-field-33 form-group has-icon "
    >
      <label
        className={undefined}
        htmlFor="state"
      >
        State
         
        <sup>
          *
        </sup>
      </label>
      <div
        className="form-control-wrapper"
      >
        <select
          className="form-control select-control"
          id="state"
          name="state"
          onBlur={[Function]}
          onChange={[Function]}
          value=""
        >
          <option
            value=""
          >
            â–¾ Select a state
          </option>
          <option
            value="AL"
          >
            Alabama
          </option>
          <option
            value="AK"
          >
            Alaska
          </option>
          <option
            value="AS"
          >
            American Samoa
          </option>
          <option
            value="AZ"
          >
            Arizona
          </option>
          <option
            value="AR"
          >
            Arkansas
          </option>
          <option
            value="AF"
          >
            Armed Forces Africa
          </option>
          <option
            value="AA"
          >
            Armed Forces Americas
          </option>
          <option
            value="AC"
          >
            Armed Forces Canada
          </option>
          <option
            value="AE"
          >
            Armed Forces Europe
          </option>
          <option
            value="AM"
          >
            Armed Forces Middle East
          </option>
          <option
            value="AP"
          >
            Armed Forces Pacific
          </option>
          <option
            value="CA"
          >
            California
          </option>
          <option
            value="CO"
          >
            Colorado
          </option>
          <option
            value="CT"
          >
            Connecticut
          </option>
          <option
            value="DE"
          >
            Delaware
          </option>
          <option
            value="DC"
          >
            District Of Columbia
          </option>
          <option
            value="FM"
          >
            Federated States Of Micronesia
          </option>
          <option
            value="FL"
          >
            Florida
          </option>
          <option
            value="GA"
          >
            Georgia
          </option>
          <option
            value="GU"
          >
            Guam
          </option>
          <option
            value="HI"
          >
            Hawaii
          </option>
          <option
            value="ID"
          >
            Idaho
          </option>
          <option
            value="IL"
          >
            Illinois
          </option>
          <option
            value="IN"
          >
            Indiana
          </option>
          <option
            value="IA"
          >
            Iowa
          </option>
          <option
            value="KS"
          >
            Kansas
          </option>
          <option
            value="KY"
          >
            Kentucky
          </option>
          <option
            value="LA"
          >
            Louisiana
          </option>
          <option
            value="ME"
          >
            Maine
          </option>
          <option
            value="MH"
          >
            Marshall Islands
          </option>
          <option
            value="MD"
          >
            Maryland
          </option>
          <option
            value="MA"
          >
            Massachusetts
          </option>
          <option
            value="MI"
          >
            Michigan
          </option>
          <option
            value="MN"
          >
            Minnesota
          </option>
          <option
            value="MS"
          >
            Mississippi
          </option>
          <option
            value="MO"
          >
            Missouri
          </option>
          <option
            value="MT"
          >
            Montana
          </option>
          <option
            value="NE"
          >
            Nebraska
          </option>
          <option
            value="NV"
          >
            Nevada
          </option>
          <option
            value="NH"
          >
            New Hampshire
          </option>
          <option
            value="NJ"
          >
            New Jersey
          </option>
          <option
            value="NM"
          >
            New Mexico
          </option>
          <option
            value="NY"
          >
            New York
          </option>
          <option
            value="NC"
          >
            North Carolina
          </option>
          <option
            value="ND"
          >
            North Dakota
          </option>
          <option
            value="MP"
          >
            Northern Mariana Islands
          </option>
          <option
            value="OH"
          >
            Ohio
          </option>
          <option
            value="OK"
          >
            Oklahoma
          </option>
          <option
            value="OR"
          >
            Oregon
          </option>
          <option
            value="PW"
          >
            Palau
          </option>
          <option
            value="PA"
          >
            Pennsylvania
          </option>
          <option
            value="PR"
          >
            Puerto Rico
          </option>
          <option
            value="RI"
          >
            Rhode Island
          </option>
          <option
            value="SC"
          >
            South Carolina
          </option>
          <option
            value="SD"
          >
            South Dakota
          </option>
          <option
            value="TN"
          >
            Tennessee
          </option>
          <option
            value="TX"
          >
            Texas
          </option>
          <option
            value="UT"
          >
            Utah
          </option>
          <option
            value="VT"
          >
            Vermont
          </option>
          <option
            value="VI"
          >
            Virgin Islands
          </option>
          <option
            value="VA"
          >
            Virginia
          </option>
          <option
            value="WA"
          >
            Washington
          </option>
          <option
            value="WV"
          >
            West Virginia
          </option>
          <option
            value="WI"
          >
            Wisconsin
          </option>
          <option
            value="WY"
          >
            Wyoming
          </option>
        </select>
        
        
      </div>
    </div>
    <div
      className="wvus-field-zip wvus-field-33 wvus-field--end form-group has-icon "
      data-field-container="zip"
      data-field-is-valid={false}
    >
      <label
        htmlFor="zip"
      >
        Zip
         
        <sup>
          *
        </sup>
      </label>
      <div
        className="form-control-wrapper"
      >
        <input
          className="form-control"
          id="zip"
          name="zip"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder={undefined}
          type="text"
          value=""
        />
        
        
        
      </div>
    </div>
  </form>
</div>
`;
